x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { run } from '@kbn/dev-cli-runner';
+import yargs from 'yargs';
+import _ from 'lodash';
+import globby from 'globby';
+import pMap from 'p-map';
+import { ToolingLog } from '@kbn/tooling-log';
+import { withProcRunner } from '@kbn/dev-proc-runner';
+import cypress from 'cypress';
+import { findChangedFiles } from 'find-cypress-specs';
+import grep from '@cypress/grep/src/plugin';
+import crypto from 'crypto';
+import fs from 'fs';
+import { createFailError } from '@kbn/dev-cli-errors';
+import axios from 'axios';
+import path from 'path';
+import os from 'os';
+import { renderSummaryTable } from './print_run';
+import type { SecuritySolutionDescribeBlockFtrConfig } from './utils';
+import { isSkipped, parseTestFileConfig } from './utils';
+
+interface ProductType {
```

Comment:
NIT: Let's use the `SecurityProductTypes` type already created instead of creating a new one.

Reply:
Tried to address my comment but we ended up creating circular dependency. In a follow-up PR we should consider moving the scripts outside the security solution folder as we did with our cypress tests, since we did it with the aim of reusing code without the risk of creating circular dependencies.

These were the approaches:
* [Approach 1](https://github.com/elastic/kibana/pull/169311/commits/2fd9d9fecad90b4f41455b1eaf0c8ca63186c770) $\rightarrow$ [result](https://buildkite.com/elastic/kibana-pull-request/builds/172240#018b860b-ea49-456a-8a46-c3b4f44cf4ed)

* [Approach 2](https://github.com/elastic/kibana/pull/169311/commits/ac7b53fb83e82d5bee7e212d430934493201c767) $\rightarrow$ [result](https://buildkite.com/elastic/kibana-pull-request/builds/172246#018b8619-7a33-4737-9573-57f1aba08cde)

Reply:
Addressed with commit https://github.com/elastic/kibana/pull/169311/commits/9e33e810a94dc815058b54ffad109a431ce9f458

Reply:
Changes in commit Addressed with commit https://github.com/elastic/kibana/pull/169311/commits/9e33e810a94dc815058b54ffad109a431ce9f458 have been reverted. We will address it in another PR.